### PR TITLE
Improve efficiency by using entrySet over keySet

### PR DESCRIPTION
**Pull request by KernelDeimos**
This is more efficient because the map gets queried once per key rather than twice.

The change turned out to be much easier than I expected because it exists in code generation.

### CODE REVIEW

File: src/foam/java/refinements.js [hunk 1]
```diff
@@ -1531,12 +1531,12 @@ foam.CLASS({
...
-          map.put(getName() + "." + key, new foam.lib.csv.FromCSVSetter() {
+        for ( java.util.Map.Entry<String, foam.lib.csv.FromCSVSetter> entry : map2.entrySet() ) {
+          map.put(getName() + "." + entry.getKey(), new foam.lib.csv.FromCSVSetter() {
             public void set(foam.core.FObject obj, String str) {
               try {
                 if ( prop.get(obj) == null ) prop.set(obj, prop.of().newInstance());
-                map2.get(key).set((foam.core.FObject) prop.get(obj), str);
+                entry.getValue().set((foam.core.FObject) prop.get(obj), str);
               } catch ( Throwable t ) {
                 // ???
```
Comment (jlhughes) [inline src/foam/java/refinements.js]: It would be nice to log this 'debug' if possible.

Comment (KernelDeimos) [reply src/foam/java/refinements.js]: Oh wow, my eyes skipped right over that thing. I can see why someone put question marks there though - PropertyInfo isn't ContextAware, but it does have access to ClassInfo but that's not ContextAware either... maybe conditionally log if the class has a Property axiom `x` of type `Context`?

Comment (jlhughes) [reply src/foam/java/refinements.js]: Well think only a System.err makes sense, as this code will be called during log message creation.

Comment (jlhughes) [inline src/foam/java/refinements.js]:
```suggestion
                t.printStackTrace() // cannot use logging from logging. 
```

**Review by jlhughes — APPROVED** [review]

**Pull request merged**